### PR TITLE
HPCC-30131 Cloud: Support HPCC Remote Trust via shared cert authority

### PR DESCRIPTION
**Pull request by afishbeck**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: roxie/ccd/ccdprotocol.cpp [hunk 1]
```diff
@@ -228,10 +228,11 @@ class ProtocolSocketListener : public ProtocolListener
     StringAttr keyFile;
     StringAttr passPhrase;
     Owned<ISecureSocketContext> secureContext;
+    Owned<IPropertyTree> tlsConfig;
```
Comment (ghalliday) [inline roxie/ccd/ccdprotocol.cpp:231]: This doesn't appear to be used.

File: system/security/securesocket/securesocket.cpp [hunk 2]
```diff
@@ -1974,6 +1974,7 @@ SECURESOCKET_API ISecureSocketContext* createSecureSocketContextEx2(const IPrope
     if (config == NULL)
         return createSecureSocketContext(sockettype);
 
+    dbglogXML(config);
```
Comment (ghalliday) [inline system/security/securesocket/securesocket.cpp:1977]: Tracing left in?

File: roxie/ccd/ccdprotocol.cpp [hunk 3]
```diff
@@ -242,9 +243,16 @@ class ProtocolSocketListener : public ProtocolListener
         keyFile.set(_keyFile);
         passPhrase.set(_passPhrase);
         isSSL = streq(protocol.str(), "ssl");
+        DBGLOG("ProtocolSocketListener port=%d, certFile=%s, keyFile=%s", port, _certFile, _keyFile);
```
Comment (ghalliday) [inline roxie/ccd/ccdprotocol.cpp:246]: Double checking that this is only logged once for the process.

**Review by ghalliday — COMMENTED** [review]
@afishbeck a couple of minor comments/questions.  Please feel free to squash any changes into this PR to ease merging (but don't rebase).

File: roxie/ccd/ccdprotocol.cpp (hunk 3 above)
Comment (afishbeck) [reply roxie/ccd/ccdprotocol.cpp:246]: Yes, these should be instantiated (and log) at roxie startup time.

Comment (afishbeck) [reply roxie/ccd/ccdprotocol.cpp:246]: That being said, I don't think I meant to add this line, it doesn't even cover my change, I'll remove it.

File: system/security/securesocket/securesocket.cpp (hunk 2 above)
Comment (afishbeck) [reply system/security/securesocket/securesocket.cpp:1977]: Yes, I'll remove it.

File: roxie/ccd/ccdprotocol.cpp (hunk 1 above)
Comment (afishbeck) [reply roxie/ccd/ccdprotocol.cpp]: Removed.

File: roxie/ccd/ccdprotocol.cpp (hunk 3 above)
Comment (afishbeck) [reply roxie/ccd/ccdprotocol.cpp]: Removed.

**Comment by afishbeck** [comment]
@ghalliday updated and squashed at the same time.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**